### ISSUE TITLE
Debian packaging

### ISSUE THREAD

**Issue by aviau** [syncany/syncany#261]
Hello!

I would really like to see syncany in Debian. So I decided to start working on the Debian packaging.

You may follow my work [here](https://github.com/ReAzem/syncany-gbp) (If there is interest, we could move the repo to Alioth). My repo uses git-buildpackage.

@binwiederhier If you agree, I think it would be a great idea to move the packaging in the syncany organization. Proper packaging needs to be in a separate repo, a Debian folder won't do it. The repo should be  called something like syncany-debian or syncany-gbp.

Because syncany is still in alpha, I think we should push it in the experimental repos.

I don't have any experience in Java maintaining. So I will learn as I go. Feel free to come and help.

Before we start with syncany, I would like us to make a list of all of syncany's dependencies. Feel free to edit my post to document dependencies. If we find dependencies that are not packaged, I will open a Request for Package for each of them.

| Dependency  | Version | In Debian?  |  Bug opened |
|---|---|---|---|
|  [gradle](http://www.gradle.org/) | ???  |  [Yes](https://packages.qa.debian.org/g/gradle.html) - 1.5 | |
|[commons-io](http://commons.apache.org/proper/commons-io/)|2.4|[Yes](https://packages.qa.debian.org/c/commons-io.html) - 2.4|---|
|[bouncycastle](http://www.bouncycastle.org/)|1.51|[Yes](https://packages.qa.debian.org/b/bouncycastle.html) - 1.51|---|
|[simple-xml](http://simple.sourceforge.net/)|2.7.1|[Yes](https://packages.qa.debian.org/s/simple-xml.html) - 2.7.1 |---|
|[libguava-java](https://code.google.com/p/guava-libraries/)|15.0|[Yes](https://packages.qa.debian.org/g/guava-libraries.html) - 17.0|---|
|[commons-codec](http://commons.apache.org/proper/commons-codec/)|1.8|[Yes](https://packages.qa.debian.org/libc/libcommons-codec-java.html) - 1.9 |---|
|[hsqldb](http://hsqldb.org/)|2.3.1|[Yes](https://packages.qa.debian.org/h/hsqldb.html) - 2.2.9|---|
|[com.github.zafarkhaja:java-semver](https://github.com/zafarkhaja/jsemver)|0.7.2|No|[RFS#778686](https://bugs.debian.org/778686) / [ITP#766999](https://bugs.debian.org/766999)|
|[undertow](http://undertow.io/)|1.1.0.CR3| No |[RFP#767001](https://bugs.debian.org/767001)|
|[jpathwatch](http://jpathwatch.wordpress.com/)|0.95|[Yes](https://packages.qa.debian.org/j/jpathwatch.html) - 0.95|---|
|JUnit|4.9|[Yes](https://packages.qa.debian.org/j/junit.html) - 3.8.2||

**Comment by ReAzem** [comment]
I have begun working on jsemver [here](http://anonscm.debian.org/cgit/pkg-java/jsemver.git). There is no rush as a new package cannot get into Debian during the freeze, so it is best to work on existing packages.

When the freeze ends, I should be able to upload jsemver and hopefully someone manifests interest to package undertow. If not, I'll do it.

**Comment by binwiederhier** [comment]
:+1:

**Comment by ReAzem** [comment]
My jsemver package is done and I have opened a Request for Sponsorship. I have updated the table in the first post of this issue.

**Comment by binwiederhier** [comment]
FYI: Gson & HttpComponents were added.
https://github.com/syncany/syncany/issues/33#issuecomment-96435446

**Comment by aviau** [comment]
[Syncany is now in Debian](https://packages.qa.debian.org/s/syncany.html)

**Comment by binwiederhier** [comment]
Wowwwww!!!!!!! Niceeeeeeeee!!!!!

**Comment by binwiederhier** [comment]
Seems to work nicely:

```bash
# /etc/apt/sources.list.d/syncany.list
deb http://httpredir.debian.org/debian experimental main
deb http://httpredir.debian.org/debian unstable main

# Install
apt-get update; apt-get install -t experimental syncany
```

**Comment by aviau** [comment]
I don't know if apt pinning is required for experimental, but here is my preferences file.

I have all repos in my lists: stable, testing, unstable, experimental. This preferences file sets my preference to testing. Unless I say otherwise using ``-t unstable`` for example, it will install everything from testing.
```
# /etc/apt/preferences

Package: *
Pin: release a=testing
Pin-Priority: 900

Package: *
Pin: release o=Debian
Pin-Priority: -10

```
